lib/streamaggr/streamaggr.go

```diff
@@ -1210,26 +1211,37 @@ func (ctx *flushCtx) flushSeries() {
 	}
 
 	// Slow path - apply output relabeling and then push the output metrics.
-	auxLabels := promutil.GetLabels()
-	dstLabels := auxLabels.Labels[:0]
 	dst := tss[:0]
+	wg := sync.WaitGroup{}
+	dstCh := make(chan *prompb.TimeSeries, len(tss))
+	rSem := make(chan struct{}, runtime.NumCPU())
+
 	for _, ts := range tss {
-		dstLabelsLen := len(dstLabels)
-		dstLabels = append(dstLabels, ts.Labels...)
-		dstLabels = outputRelabeling.Apply(dstLabels, dstLabelsLen)
-		if len(dstLabels) == dstLabelsLen {
-			// The metric has been deleted by the relabeling
-			continue
-		}
-		ts.Labels = dstLabels[dstLabelsLen:]
-		dst = append(dst, ts)
+		rSem <- struct{}{}
+		wg.Add(1)
+		go func(ts *prompb.TimeSeries) {
+			defer wg.Done()
+			dstLabels := make([]prompb.Label, 0, len(ts.Labels))
```

Comment:
why don't you use `promutil.GetLabels()` here?

Reply:
No real strong reason, I'll switch it over